github/resource_github_branch_protection_test.go

```diff
@@ -36,6 +36,9 @@ func TestAccGithubBranchProtection_basic(t *testing.T) {
 					resource.TestCheckResourceAttr(rn, "repository", repoName),
 					resource.TestCheckResourceAttr(rn, "branch", "master"),
 					resource.TestCheckResourceAttr(rn, "enforce_admins", "true"),
+					resource.TestCheckResourceAttr(rn, "allow_force_pushes", "true"),
+					resource.TestCheckResourceAttr(rn, "allow_deletions", "true"),
+					resource.TestCheckResourceAttr(rn, "require_linear_history", "true"),
```

Comment:
Nit: a separate test, focused on setting these optional attributes, would be helpful to validate these attributes instead of adding to this `basic` test